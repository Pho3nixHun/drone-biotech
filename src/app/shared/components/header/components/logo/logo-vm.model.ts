export interface LogoVM {
    imageSrc: string;
    altText: string;
}
