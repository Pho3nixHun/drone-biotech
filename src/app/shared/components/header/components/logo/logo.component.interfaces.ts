export interface LogoComponentInterfaces {
  imageSrc: string;
  routerLink: string | string[];
  altText: string;
}
