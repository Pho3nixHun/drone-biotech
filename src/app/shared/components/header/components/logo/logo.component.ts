import { Component, input } from '@angular/core';
import { LogoComponentInterfaces } from '@components/header/components/logo/logo.component.interfaces';

@Component({
  selector: 'app-logo',
  standalone: true,
  imports: [],
  templateUrl: './logo.component.html',
})
export class LogoComponent {
  vm = input<LogoComponentInterfaces | null>(null, { alias: 'vm' });
}
