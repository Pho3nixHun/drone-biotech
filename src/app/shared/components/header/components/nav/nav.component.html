<nav
  class="flex flex-col items-center space-y-4 text-center md:items-start md:flex-row md:space-y-0 md:space-x-10 md:text-left"
>
  <ng-content select="app-nav-item" />
</nav>
