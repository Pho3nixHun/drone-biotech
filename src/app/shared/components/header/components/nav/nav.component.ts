import { Component } from '@angular/core';

@Component({
  selector: 'app-nav',
  standalone: true,
  imports: [],
  templateUrl: './nav.component.html',
})
export class NavComponent {}
