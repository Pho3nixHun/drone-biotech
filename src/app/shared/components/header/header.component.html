<section class="p-4 flex justify-center">
  <div
    class="w-full space-y-4 flex flex-col justify-between items-center md:w-3/4 md:flex-row md:space-y-0"
  >
    <ng-content select="[slot=logo]"></ng-content>
    <ng-content select="[slot=nav]"></ng-content>
  </div>
</section>
