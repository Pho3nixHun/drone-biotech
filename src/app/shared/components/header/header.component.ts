import { Component } from '@angular/core';
import { HeroComponent } from '@components/hero/hero.component';
import { ScrollDirective } from '@directives/scroll.directive';

@Component({
  selector: 'app-header',
  standalone: true,
  imports: [HeroComponent, ScrollDirective],
  templateUrl: './header.component.html',
})
export class HeaderComponent {}
