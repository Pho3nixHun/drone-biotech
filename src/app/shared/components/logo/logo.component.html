<a href="">
  <img [src]="imageSrc" alt="logo" class="w-12 h-12 md:w-16 md:h-16 mb-4 md:mb-0" />
</a>
