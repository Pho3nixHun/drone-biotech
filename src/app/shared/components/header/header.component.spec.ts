import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HeaderComponent } from './header.component';
import { NavComponent } from '@components/header/components/nav/nav.component';
import { Component } from '@angular/core';
import { LogoComponent } from '@components/header/components/logo/logo.component';

@Component({
  template: `
    <app-header>
      <app-logo class="testlogo" />
      <app-logo class="testlogo" />
      <app-nav class="testnav">app-nav element 1</app-nav>
      <app-nav class="testnav">app-nav element 2</app-nav>
      <div class="testdiv">This should not be projected</div>
    </app-header>
  `,
})
class TestHostComponent {}
describe('HeaderComponent', () => {
  let component: TestHostComponent;
  let fixture: ComponentFixture<TestHostComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [HeaderComponent, LogoComponent, NavComponent],
      declarations: [TestHostComponent],
    }).compileComponents();
    fixture = TestBed.createComponent(TestHostComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.debugElement.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  //Snapshot test
  it('should project <app-logo>, <app-nav> elements and ignore other elements', () => {
    // Arrange

    // Act

    // Assert
    expect(compiled.innerHTML).toMatchSnapshot();
  });
});
