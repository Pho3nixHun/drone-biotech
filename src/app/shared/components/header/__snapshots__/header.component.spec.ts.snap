// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should project <app-logo>, <app-nav> elements and ignore other elements 1`] = `
<div
  id="root0"
  ng-version="18.2.8"
>
  <app-header>
    <section
      class="p-4 flex justify-center"
    >
      <div
        class="w-full space-y-4 flex flex-col justify-between items-center md:w-3/4 md:flex-row md:space-y-0"
      >
        <app-logo
          ng-reflect-vm="[object Object]"
        >
          <a
            href="link"
          >
            <img
              alt="lepke"
              class="w-12 h-12 md:w-16 md:h-16 mb-4 md:mb-0"
              src="assets/lepke.jpg"
            />
          </a>
        </app-logo>
        <app-logo
          ng-reflect-vm="[object Object]"
        >
          <a
            href="link"
          >
            <img
              alt="lepke"
              class="w-12 h-12 md:w-16 md:h-16 mb-4 md:mb-0"
              src="assets/lepke.jpg"
            />
          </a>
        </app-logo>
        <app-nav>
          <nav
            class="flex flex-col items-center space-y-4 text-center md:items-start md:flex-row md:space-y-0 md:space-x-10 md:text-left"
          />
        </app-nav>
        <app-nav>
          <nav
            class="flex flex-col items-center space-y-4 text-center md:items-start md:flex-row md:space-y-0 md:space-x-10 md:text-left"
          />
        </app-nav>
      </div>
    </section>
  </app-header>
</div>
`;
